### PR TITLE
Update otel version

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: dependencyManagement/build.gradle.kts [hunk 1]
```diff
@@ -7,13 +7,18 @@ data class DependencySet(val group: String, val version: String, val modules: Li
...
+
 val DEPENDENCY_BOMS = listOf(
   "com.fasterxml.jackson:jackson-bom:2.15.2",
   "com.google.guava:guava-bom:32.1.2-jre",
   "com.linecorp.armeria:armeria-bom:1.25.2",
   "org.junit:junit-bom:5.10.0",
   "io.grpc:grpc-bom:1.58.0",
-  "io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:1.28.0-alpha",
+  // even though the stable instrumentation bom is included by the alpha bom,
+  // it is specified here because Renovate doesn't update "-alpha" artifacts
```
Comment (trask) [inline dependencyManagement/build.gradle.kts:19]: this is why Renovate hasn't been updating the version in this repo

File: aws-xray/build.gradle.kts [hunk 2]
```diff
@@ -45,3 +45,10 @@ testing {
     }
   }
 }
+
+configurations.all {
+  resolutionStrategy {
+    // TODO this module still needs to be updated to the latest semconv
+    force("io.opentelemetry:opentelemetry-semconv:1.28.0-alpha")
```
Comment (trask) [inline aws-xray/build.gradle.kts:52]: need to open tracking issue

File: maven-extension/build.gradle.kts [hunk 3]
```diff
@@ -60,3 +60,10 @@ tasks {
 }
 
 tasks.getByName("test").dependsOn("shadowJar")
+
+configurations.all {
+  resolutionStrategy {
+    // TODO this module still needs to be updated to the latest semconv
+    force("io.opentelemetry:opentelemetry-semconv:1.28.0-alpha")
```
Comment (trask) [inline maven-extension/build.gradle.kts:67]: need to open tracking issue

**Review by laurit — APPROVED** [review]

File: aws-xray/build.gradle.kts (hunk 2 above)
Comment (trask) [reply aws-xray/build.gradle.kts:52]: #1043

File: maven-extension/build.gradle.kts (hunk 3 above)
Comment (trask) [reply maven-extension/build.gradle.kts:67]: #1044

**Pull request merged**